src/sql/workbench/services/connection/browser/connectionWidget.ts

```diff
@@ -282,7 +282,7 @@ export class ConnectionWidget extends lifecycle.Disposable {
 	private addDatabaseOption(): void {
 		// Database
 		let databaseOption = this._optionsMaps[ConnectionOptionSpecialType.databaseName];
-		if (databaseOption) {
+		if (databaseOption && !databaseOption.hide) {
```

Comment:
It seems like if we're going to support things not having databases (or other such fields) then we should be doing that in a more general way. For example - this is still going to result in the query editor showing the database selector right? That seems odd given what you said about these not having "databases".

So instead of just having people have to specify "hide" I would rather we made it so the code could correctly handle this option not being there. What happens if you remove it completely? Because this code seems to be able to handle that case just fine (if the options map doesn't have it then it won't add the option)

Reply:
The code handles it fine except for the query window. I haven't found any other issues beyond that so far.